umbra-js/package.json

```diff
@@ -23,7 +23,7 @@
   "author": "Matt Solomon <matt@mattsolomon.dev>, Ben DiFrancesco <ben@scopelift.co>",
   "license": "ISC",
   "dependencies": {
-    "@unstoppabledomains/resolution": "3.0.0",
+    "@unstoppabledomains/resolution": "^7.1.4",
```

Comment:
_(arbitrary comment location to start a thread)_

Is this below portion of the code still correct? https://github.com/ScopeLift/umbra-protocol/blob/5f5cb883d2d4d6b1cf0caed994d51c2477a04ba5/frontend/src/utils/address.ts#L40-L64

More generally, we are deployed on mainnet, polygon, arbitrum, and optimism—how should name resolution work on each network, given that IIUC domains are minted on polygon? We need to be able to resolve names in both directions: from an address to name (when a user connects their wallet, so it shows in the top right instead of an address) and from a name to an address.

Related to that, can a name resolve to a contract, and if so how should we handle that? For example, right now all ENS name resolution is done with a mainnet provider. If someone tries withdrawing funds to `vitalik.eth`, and that resolves to a contract, we warn the user because that contract may not exist on the current network, and even if it does, it might not have the same state/owners (e.g. if it's a multisig). Relevant code here:

https://github.com/ScopeLift/umbra-protocol/blob/5f5cb883d2d4d6b1cf0caed994d51c2477a04ba5/frontend/src/utils/address.ts#L132-L141

Reply:
The domains are minted on polygon, but the library resolves to a mainnet address. The library first looks for a domain on Polygon mainnet, and if it's not found there than it will check on Ethereum mainnet.

I'm afraid I don't know if the code in frontend/src/utils/address.ts is still correct and I'm couldn't find in the docs if a name can resolve to a contract. I will ask for someone more experienced with the library so you can update the code if needed.

Reply:
To be honest I'm still not sure I understand the architecture and how that maps to what we need to do for forward and reverse resolution. I'm sure I can find these answers in the docs, but just laying out some unanswered questions in this thread as a reminder for myself to look into once I have the time (or if you know the answers / have links to the docs that would also help a lot).

1. What is the the process for resolving a name to an address? Based on the PR is looks like this is still `await resolution.addr(name, 'ETH')`, but I'm not sure if always passing `ETH` is correct for all networks we support.

2. What is the the process for reverse-resolving an address to a name? Previously we used a subgraph to look up all names owned by an address and just chose the first name returned in that array. I'm not sure if that's still considered the correct way.

3. Given that an address can resolve to a contract, let's confirm the current approach used for ENS of "warn the user because that contract may not exist on the current network, and even if it does, it might not have the same state/owners (e.g. if it's a multisig)" is the correct approach.

4.  Clarification around this behavior: "The domains are minted on polygon, but the library resolves to a mainnet address. The library first looks for a domain on Polygon mainnet, and if it's not found there than it will check on Ethereum mainnet." — I don't fully understand this one, and I'm not sure what this means from a developer standpoint. 
    1. Why does it always default to polygon
    2. Why does it look on mainnet if domains are minted on polygon?
    3. How does being connected to e.g. Optimism/Arbitrum affect how we should handle domain resolution?